src/tiling/tiler.rs

```diff
@@ -73,17 +74,37 @@ impl TilingInfo {
     let min_tiles_log2 = min_tile_cols_log2
       .max(Self::tile_log2(max_tile_area_sb, sb_cols * sb_rows));
 
-    let tile_cols_log2 =
-      tile_cols_log2.max(min_tile_cols_log2).min(max_tile_cols_log2);
-    let tile_width_sb = sb_cols.align_power_of_two_and_shift(tile_cols_log2);
+    let mut tile_cols_log2 =
+      tile_cols_log2_requested.max(min_tile_cols_log2).min(max_tile_cols_log2);
 
     let min_tile_rows_log2 = if min_tiles_log2 > tile_cols_log2 {
       min_tiles_log2 - tile_cols_log2
```

Comment:
`min_tile_rows_log2` is used to write the expected bitstream: https://github.com/xiph/rav1e/blob/a38059cf28bd75f7d42c9defee071a2b71d2c3c1/src/header.rs#L633

I did not check in detail, but here, it depends on `tile_cols_log2`, and `tile_cols_log2` is modified afterwards (below). So I suspect that under some conditions, `min_tile_rows_log2` may end up inconsistent with `tile_cols_log2`, leading to bitstream corruption.

Reply:
Btw, the line I quoted is incorrect:

```
diff --git a/src/header.rs b/src/header.rs
index c863a32c..502e6e54 100644
--- a/src/header.rs
+++ b/src/header.rs
@@ -630,7 +630,7 @@ impl<W: io::Write> UncompressedHeader for BitWriter<W, BigEndian> {
       self.write_bit(false);
     }
 
-    let rows_ones = ti.tile_rows_log2 - ti.min_tile_cols_log2;
+    let rows_ones = ti.tile_rows_log2 - ti.min_tile_rows_log2;
     for _ in 0..rows_ones {
       self.write_bit(true);
     }
```

Reply:
Indeed. I'll move the calculation of this to after the loop. (I think it breaks >4k wide video)

Reply:
I had also considered an alternate implementation that just creates entire new TilingInfos in a loop instead, if you think that would be cleaner.